tests/test_projectgen_oids.py

```diff
@@ -337,6 +346,27 @@ def _oids_tids_none(self, generator, strategy):
                 assert (
                     default_value_definition.expression() == expected_other_expression
                 )
+                # check if not null and unique constraints are set
+                assert (
+                    t_ili_tid_field.constraints().constraints()
+                    & QgsFieldConstraints.ConstraintUnique
+                )
+                assert (
+                    t_ili_tid_field.constraints().constraints()
+                    & QgsFieldConstraints.ConstraintNotNull
+                )
+                assert (
+                    t_ili_tid_field.constraints().constraintStrength(
+                        QgsFieldConstraints.ConstraintUnique
+                    )
+                    == QgsFieldConstraints.ConstraintStrengthHard
+                )
+                assert (
+                    t_ili_tid_field.constraints().constraintStrength(
+                        QgsFieldConstraints.ConstraintNotNull
+                    )
+                    == QgsFieldConstraints.ConstraintStrengthHard
+                )
```

Comment:
Since the same block is repeated several of times, it would be worth adding a `_check_t_ili_tid_field_constraints()` method doing these checks.